packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -31,6 +31,11 @@ import {
 import { getAlignment } from './alignment';
 import { BasePlacement, Placement, PlacementWithAlignment } from './placement';
 
+/**
+ * @internal
+ */
+export type triggerEvent = 'click' | 'hover' | 'focus';
```

Comment:
Can you please validate if the Event is really exposed corrently (Internal behavior)? Normally it should be a feature from stenciljs 3 https://stenciljs.com/docs/introduction/upgrading-to-stencil-three#custom-types-for-props-and-events-are-now-exported-from-componentsdts
If not you have to place it in a seperate file outside of the component file.

Other point is types should start with an uppercase char `DropdownTriggerEvent` and to avoid possible name clonflicts with other Types i would suggest to call it `DropdownTriggerEvent` (https://stenciljs.com/docs/style-guide#html-tag)